seth-shaw-unlv/controlled_access_terms
# Issue 1561: Remove dependency on Name module

**GitHub Issue**: https://github.com/Islandora/documentation/issues/1561

Alternate to @rosiel's PR: #53 

# What does this Pull Request do?

Removes the dependency on the Name module (they will only see the alternate name string field on a fresh islandora install), but keeps the related fields around in case someone wants to install the module themselves. (Prevents breaking new [ArchivesSpace Module](https://drupal.org/project/archivesspace) installs which use these fields.)

# What's new?

*. Moves the person preferred name and alternate name fields from install to optional.
* Adds a new alternate name (string) field.
* Removes the dependency on the name module.
* Moves the name module in composer to the suggested section with a note that it provides for structured western-centric names.

# How should this be tested?

* Take a fresh islandora
* Delete the prefered and alternate name fields from the person taxonomy.
* Apply the PR and clear cache.
* Uninstall (`http://localhost:8000/admin/modules/uninstall`) the Name module. _(That is the primary requirement.)_
* Import the controlled access terms defaults feature (`http://localhost:8000/admin/config/development/features/diff/controlled_access_terms_defaults`)
* Check the Person taxonomy that only the new string-based alternate name string field is available.

# Additional Comments

I figured it would be easier to just go ahead and make a fresh PR rather than making @rosiel slog through the manual edits again. Let me know if there is anything I missed.

# Interested parties
@Islandora/8-x-committers, especially @rosiel.
